src/wolframite/tools/graphics.clj

```diff
@@ -34,8 +34,16 @@
   @default-app)
 
 (defn- fit-graphic-expr-to-frame ^String [^String wl-expr-str ^Component container]
-  (let [size (.getSize container)]
-    (str "Show[" wl-expr-str ", ImageSize -> {" (.width size) "," (.height size) "}]")))
+  (let [parent (.getParent container)
+        size (if (instance? JViewport parent)
+               (.getExtentSize ^JViewport parent)
```

Comment:
ExtentSize is typically smaller thant size (b/c of the scrollbards, I assume?) so let's use that.